base_layer/wallet/src/key_manager_service/storage/sqlite_db/key_manager_state.rs

```diff
@@ -183,17 +186,20 @@ impl Encryptable<XChaCha20Poly1305> for NewKeyManagerStateSql {
             .to_vec()
     }
 
-    fn encrypt(&mut self, cipher: &XChaCha20Poly1305) -> Result<(), String> {
-        self.primary_key_index = encrypt_bytes_integral_nonce(
+    fn encrypt(self, cipher: &XChaCha20Poly1305) -> Result<Self, String> {
+        let mut output = self.clone();
+        output.primary_key_index = encrypt_bytes_integral_nonce(
```

Comment:
It's so strange to encrypt the key index. Is this really necessary? What's the danger in leaking this?

Reply:
I have no idea, tbh. I think this is legacy code